R/bert_classification-predict.R

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 Bedford Freeman & Worth Pub Grp LLC DBA Macmillan Learning.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#' Predict from a `bert_classification`
+#'
+#' @param object A `bert_classification` object.
+#'
+#' @param new_data A data frame or matrix of new predictors.
+#'
+#' @param type A single character. The type of predictions to generate.
+#' Valid options are:
+#'
+#' - `"class"` for "hard" class predictions.
+#' - `"prob"` for class probabilities. (NOT IMPLEMENTED AT ALL YET)
+#'
+#' @param ... Not used, but required for extensibility.
+#'
+#' @return
+#'
+#' A tibble of predictions. The number of rows in the tibble is guaranteed
+#' to be the same as the number of rows in `new_data`.
+#'
+#' @export
+predict.bert_classification <- function(object,
+                                        new_data,
+                                        type = "class",
+                                        ...) {
+  forged <- hardhat::forge(new_data, object$blueprint)
```

Comment:
Again, I'm going to have to learn more about hardhat before I fully understand this. I've skimmed the vignettes, but there's a lot to absorb. :)